pkg/provisioner/provisioner.go

```diff
@@ -74,23 +74,29 @@ type PreprovisioningImage struct {
 }
 
 type ManagementAccessData struct {
-	BootMode                   metal3api.BootMode
-	AutomatedCleaningMode      metal3api.AutomatedCleaningMode
-	State                      metal3api.ProvisioningState
-	CurrentImage               *metal3api.Image
-	PreprovisioningImage       *PreprovisioningImage
-	PreprovisioningNetworkData string
-	HasCustomDeploy            bool
-	DisablePowerOff            bool
-	CPUArchitecture            string
+	BootMode                         metal3api.BootMode
+	AutomatedCleaningMode            metal3api.AutomatedCleaningMode
+	State                            metal3api.ProvisioningState
+	CurrentImage                     *metal3api.Image
+	PreprovisioningImage             *PreprovisioningImage
+	PreprovisioningNetworkData       string
+	PreprovisioningExtraKernelParams string
```

Comment:
It's a bit unfortunate that we can't use the ExtraKernelParams field inside `PreprovisioningImage.GeneratedImage`, but when no image is set then `PreprovisioningImage` has to be `nil`, and I don't see a way around that.
I wonder if we should stop inheriting from the `imageprovider.GeneratedFormat` struct - which we were only really using for convenience - here though, so that we don't have two copies of the field with different values.